src/main/java/io/cdap/plugin/gcp/common/GCPErrorDetailsProviderUtil.java

```diff
@@ -82,8 +104,11 @@ public static ProgramFailureException getHttpResponseExceptionDetailsFromChain(T
       if (t instanceof HttpResponseException) {
```

Comment:
On a second thought, since `HttpResponseException` has more details then `ApiException`, can you please traverse the `causalChain` in descending order?

```
 List<Throwable> causalChain = Throwables.getCausalChain(exception);
for (int i = causalChain.size() - 1; i >= 0; i--) {
  // do something.......
}
```